### ISSUE TITLE
MiniKeePass no longer appears in iOS "Open With" list

### ISSUE THREAD

**Issue by gvoysey** [MiniKeePass/MiniKeePass#428]
running iOS 9.3 and MiniKeePass 1.6.2, i cannot load a `*.kdbx` file from the Google Drive app (4.7.40373).  The option to "Open With" or "Send to" does not list MiniKeePass as an option.

**Comment by bmdead** [comment]
Same here... please provide solution as this is important.

**Comment by richardnicholson** [comment]
Same problem opening from Dropbox 6.2, iOS 9.3.

**Comment by realdubb** [comment]
Known issue - It's a Google Drive issue and there is a workaround
https://github.com/MiniKeePass/MiniKeePass/issues/376

With Dropbox - make sure the extension .kdbx is there and it should work

**Comment by GunterVenter** [comment]
The new Update MiniKeePass 1.6.3 too in iOS 9.3.1, cannot load a *.kdbx file from the option "Open With" or "Send to". It does not list MiniKeePass as an option. No List with OneDrive, Dropbox, GoogleDrive, EMail etc. ! Please correct this important issue.

**Comment by just6two** [comment]
I've the same issue with MiniKeePass 1.6.3 and iOS 9.3.2. cannot load file from Google Drive. Reading all that and other topics, I made a test: 
I created a database with the name **test.kdbx.kdbx** with the idea if one extention will get lost, the second maybe can activate the decoding of the required application. looks good with my test file. Can open it as ever. for me it's an option while I'm waiting for the correction...

**Comment by Goofi58** [comment]
Thank you for your comment. I had testet the same Idea, and it works also with my other Apps. It seems to be a problem with the cooperation between Keepass and iOS. All Apps lost the first Extension kdbx.

**Comment by EagleTRL57** [comment]
I will have to try this workaround.  Developer keeps blaming iOS, but interesting that none of my other apps have this problem.

**Comment by gvoysey** [comment]
The workaround suggested by @just6two works for me.  It's a hack, though.

**Comment by EagleTRL57** [comment]
Yes, a hack, but as @just6two said, it is at least an option until the developers figure out the real issue and fix it.  I am not a programmer, but figure there has to be something in the way iOS recognizes the extensions - maybe that changed?

Regardless, I guess I will keep MiniKeePass and KeePass a little longer.

**Comment by vasylenko** [comment]
@just6two many thanks for your workaround! **It does work**!!! 
`iOS 9.3.2, MiniKeePass 1.6.3`

**Comment by Dazag** [comment]
It's happening again I'm afraid of. I'm on iOS 9.3.5 iphone

**Comment by Dazag** [comment]
My bad, there was no .kdbx at the end of the file name. Just added it and worked again.

**Comment by stevecoh1** [comment]
An issue for me too with IPhone IOS 10.2 and dropbox.  Dropbox sees the files and doesn't know what to open them with. renaming the file to *.kdbx.kdbx does help- thanks.  When will this be fixed???

**Comment by vasylenko** [comment]
It looks like *.kdbx.kdbx solution works well only for Google Drive

**Comment by stevecoh1** [comment]
No, it works for dropbox, as I said above.

**Comment by laurie-hedge** [comment]
Unfortunately, the double extension work around isn't working for me :(. I'm on iOS 10.3.

**Comment by EagleTRL57** [comment]
So, the problem appears to be back and neither the single nor double .kdbx extension is working.  I have tried with Dropbox and Google Drive.  Quite frustrating.  MiniKeePass is pretty much useless to me, which means KeePass isn't working either.

**Comment by Goofi58** [comment]
The *.kdbx.kdbx solution works well with my iPhone6s, iOS 10.3.1 and Google Drive
You must try "open in" in GoogleDrive, than it works well

**Comment by laurie-hedge** [comment]
Well, I'm glad it's still working for some people, but this is what I'm getting on my iPad. As you can see, there's no MiniKeePass...
![img_0001](https://cloud.githubusercontent.com/assets/15697168/25591013/c631d5f2-2eaa-11e7-93f8-7b06da82a5a6.PNG)

**Comment by Goofi58** [comment]
Hi Laurie,
You have to klick the three dots right of the database ( not the database itself!), then choose "open in" and you can select keepass :)
Hope i could help you ;)

**Comment by Goofi58** [comment]
![img_4279](https://cloud.githubusercontent.com/assets/19272670/25595039/9b254f60-2ec3-11e7-81cf-da24c0def4f6.JPG)
![img_4280](https://cloud.githubusercontent.com/assets/19272670/25595123/e8661728-2ec3-11e7-9c90-1fa86e4e4b25.JPG)

**Comment by laurie-hedge** [comment]
Hey Goofi58. Thanks! That works great for me!

**Comment by EagleTRL57** [comment]
Hey Goofi58, thanks for posting!  I didn't realize they changed the open in options to the side of the file.  It is back working!

**Comment by EagleTRL57** [comment]
Hey, all - FWIW, I just found another KeePass compatible app - iKeePass.  It is $0.99, but natively works without the double extension, plus, you can synch automatically thru Dropbox.  Worth checking out and probably worth the hassle of copying and renaming the files.  Also, lots of good reviews on it.

**Comment by gvoysey** [comment]
Dropbox is creepy, ikeepass is closed source, and i'm cheap.  

Three strikes, right there :)

**Comment by realdubb** [comment]
Recent google drive update fixes this
https://github.com/MiniKeePass/MiniKeePass/issues/376#issuecomment-306554439

**Comment by mikob** [comment]
I had to press " open in application" in the 2nd row of options (after copy) on my iPhone.

**Comment by hnaparst** [comment]
The "Open in" option is removed from Dropbox now.  I am running iOS 11 Public Beta 2.

**Comment by sky2828** [comment]
MiniKeePass is the best security and convenience password manager in the world! But, unfortunately - the login field and the password does not work filling out on ios 9.3.5. - and for iphone 4 it is impossible to change the version of ios 9.3.5. Similar problem KeepassTouch. Dear developers can fix it? - you are welcome :)

**Comment by gvoysey** [comment]
sounds like that should be a new issue.

**Comment by jnis77diver** [comment]
Same issue opening the file from Dropbox on On iPhone 6 running iOS 11.2.1, MiniKeePass is not one of the options when you click Open With... 
Also, I tried iKeePass as suggested by @EagleTRL57 and it's a no go. Error trying to connect to Dropbox within the iKeePass app.

**Comment by Goofi58** [comment]
Hi Jonah,
you have to export the kbdx-File by clicking on it, then click on the three dots on top right ..., then select EXPORT. Then press "open in application" in the 2nd row of options and you can select MiniKeePass.
MiniKeePass now works correctly even without a double extension on iPhone6S, iOS 11.2.1 and Dropbox, GoogleDrive, OneDrive!

**Comment by Goofi58** [comment]
Hi Jonah, 
you have to export the kbdx file by clicking on it, then click on the three dots on top right ..., then select EXPORT. Then press "open in application" in the 2nd row of options and you can select MiniKeePass.
MiniKeePass now works correctly even without a double extension on iPhone6S, iOS 11.2.1 and Dropbox, GoogleDrive, OneDrive!

**Comment by jnis77diver** [comment]
@Goofi58 Yep, I had done all that. It turns out that the keepass file had a .lock extension on it that I assume Dropbox was adding which was making the file not appear like it was not a keepass file (kdbx extension) and therefore not seen as one that could be opened by MiniKeePass. I didn't notice that initially but caught it later. It may be something that other Dropbox users experience also so hopefully a worthwhile post. Thanks for responding.
